src/components/Selector/index.tsx

```diff
@@ -0,0 +1,256 @@
+import { encodeURIExtraParams } from '@app/hooks/useSearchInput';
+import type {
+  TmdbCompanySearchResponse,
+  TmdbGenre,
+  TmdbKeywordSearchResponse,
+} from '@server/api/themoviedb/interfaces';
+import type { GenreSliderItem } from '@server/interfaces/api/discoverInterfaces';
+import type { Keyword, ProductionCompany } from '@server/models/common';
+import axios from 'axios';
+import { useEffect, useState } from 'react';
+import { defineMessages, useIntl } from 'react-intl';
+import type { MultiValue, SingleValue } from 'react-select';
+import AsyncSelect from 'react-select/async';
+
+const messages = defineMessages({
+  searchKeywords: 'Search keywords…',
+  searchGenres: 'Search genres…',
+  searchStudios: 'Search studios…',
+  starttyping: 'Starting typing to search.',
+  nooptions: 'No results.',
+});
+
+type SingleVal = {
+  label: string;
+  value: number;
+};
+
+type BaseSelectorMultiProps = {
+  defaultValue?: string;
+  isMulti: true;
+  onChange: (value: MultiValue<SingleVal> | null) => void;
+};
+
+type BaseSelectorSingleProps = {
+  defaultValue?: string;
+  isMulti?: false;
+  onChange: (value: SingleValue<SingleVal> | null) => void;
+};
+
+export const CompanySelector = ({
+  defaultValue,
+  isMulti,
+  onChange,
+}: BaseSelectorSingleProps | BaseSelectorMultiProps) => {
+  const intl = useIntl();
+  const [defaultDataValue, setDefaultDataValue] = useState<
+    { label: string; value: number }[] | null
+  >(null);
+
+  useEffect(() => {
+    const loadDefaultCompany = async (): Promise<void> => {
+      if (!defaultValue) {
+        return;
+      }
+
+      const response = await axios.get<ProductionCompany>(
+        `/api/v1/studio/${defaultValue}`
+      );
+
+      const studio = response.data;
+
+      setDefaultDataValue([
+        {
+          label: studio.name ?? '',
+          value: studio.id ?? 0,
+        },
+      ]);
+    };
+
+    loadDefaultCompany();
+  }, [defaultValue]);
+
+  const loadCompanyOptions = async (inputValue: string) => {
+    if (inputValue === '') {
+      return [];
+    }
+
+    const results = await axios.get<TmdbCompanySearchResponse>(
+      '/api/v1/search/company',
+      {
+        params: {
+          query: encodeURIExtraParams(inputValue),
+        },
+      }
+    );
+
+    return results.data.results.map((result) => ({
+      label: result.name,
+      value: result.id,
+    }));
+  };
+
+  return (
+    <AsyncSelect
+      key={`company-selector-${defaultDataValue}`}
+      className="react-select-container"
+      classNamePrefix="react-select"
```

Comment:
```suggestion
      classNamePrefix="react-select"
      noOptionsMessage={({ inputValue }) =>
        inputValue === ''
          ? intl.formatMessage(messages.starttyping)
          : intl.formatMessage(messages.nooptions)
      }
```
To match the keyword selector